drupal/sites/default/boinc/modules/boincwork/includes/boincwork.helpers.inc

```diff
@@ -1470,3 +1470,357 @@ function boincwork_task_time_reported($received_time = NULL, $deadline = NULL, $
+    $sqlall .= "AND r.hostid='%s' ";
+  }
+  $sqlall .= "ORDER BY user_friendly_name";
+  $dbres_all = db_query($sqlall, $queryid);
+  db_set_active('default');
+
+  // Loop 1 of DB results
+  if ($dbres_all) {
+    while ($result = db_fetch_object($dbres_all)) {
+      $mystate = state_num($result);
+      if ( ($result->appid==$app_id) OR ($app_id==0) OR ($app_id==-1) ) {
+        $taskstates[STATE_ALL]++;
+        switch ($mystate) {
+        case STATE_IN_PROGRESS:
+          $taskstates[STATE_IN_PROGRESS]++;
+          break;
+        case STATE_PENDING:
+          $taskstates[STATE_PENDING]++;
+          break;
+        case STATE_VALID:
+          $taskstates[STATE_VALID]++;
+          break;
+        case STATE_INVALID:
+          $taskstates[STATE_INVALID]++;
+          break;
+        case STATE_ERROR:
+          $taskstates[STATE_ERROR]++;
+          break;
+        }
+      }// if app_id
+
+      //map holds a map between app ids and user friendly names for all applications.
+      $application_map[$result->appid] = $result->user_friendly_name;
+      if ( ($mystate == $tselect) OR ($tselect==STATE_ALL) ) {
+        //count of appids in the results.
+        $application_select_count[$result->appid]++;
+      }// if mystate
+
+    }// while
+  }
+  else {
+  }
+  
+  // Entry for all applications.
+  $allcount = $application_select_count ? array_sum($application_select_count) : 0;
+  $applications[-1] = bts('Application', array(), NULL, 'boinc:task-table');
+  $applications[0] = bts('All applications', array(), NULL, 'boinc:task-table') . '(' . $allcount . ')';
+  // Create application filter from application_map and application_select_count.
+  foreach($application_map as $akey => $aname) {
+    $acount = 0;
+    if ( $application_select_count and array_key_exists($akey, $application_select_count) ) {
+      $acount = $application_select_count[$akey];
+    }
+    $applications[$akey] = $aname . ' ('. $acount . ')';
+  }
+  // Header array for (sub) results table.
+  $resultheader = array(
+    array(
+      'data' => bts('Task ID', array(), NULL, 'boinc:task-table'),
+      'field' => 'id',
+    ),
+    array(
+      'data' => bts('Workunit ID', array(), NULL, 'boinc:task-table'),
+      'field' => 'workunitid',
+    ),
+    array(
+      'data' => bts('Computer', array(), NULL, 'boinc:task-table'),
+      'field' => 'hostid',
+    ),
+    array(
+      'data' => bts('Sent', array(), NULL, 'boinc:task-table')
+    ),
+    array(
+      'data' => bts('Time Reported or Deadline', array(), NULL, 'boinc:task-table')
+    ),
+    array(
+      'data' => bts('Status', array(), NULL, 'boinc:task-table')
+    ),
+    array(
+      'data' => bts('Run time', array(), NULL, 'boinc:task-table'),
+      'field'=>'elapsed_time',
+    ),
+    array(
+      'data' => bts('CPU time', array(), NULL, 'boinc:task-table'),
+      'field'=>'cpu_time',
+    ),
+    array(
+      'data' => bts('Claimed Credit', array(), NULL, 'boinc:task-table'),
```

Comment:
`claimed credit` is a remnant of the credit system before CreditNew. I concur with Shawns proposal, this way we can make sure we get every instance of this in the Drupal code.